generators/python/src/fern_python/generators/context/pydantic_generator_context_impl.py

```diff
@@ -207,3 +230,26 @@ def maybe_get_type_ids_for_type_reference(
             primitive=lambda _: None,
             unknown=lambda: None,
         )
+    
+    # Unwrap optional and alias example references
+    def unwrap_example_type_reference(
+        self, example_type_reference: ir_types.ExampleTypeReference
+    ) -> ir_types.ExampleTypeReference:
+        return example_type_reference.shape.visit(
+            primitive=lambda _: example_type_reference,
+            named=lambda named: named.shape.visit(
+                alias=lambda alias: self.unwrap_example_type_reference(alias.value),
+                enum=lambda _: example_type_reference,
+                object=lambda _: example_type_reference,
+                union=lambda _: example_type_reference,
+                undiscriminated_union=lambda _: example_type_reference,
+            ),
+            container=lambda container: container.visit(
+                list_=lambda _: example_type_reference,
+                set_=lambda _: example_type_reference,
+                optional=lambda optional: self.unwrap_example_type_reference(optional.optional) if optional.optional is not None else example_type_reference,
```

Comment:
Weird that we need the `None` check at all - when would `optional.optional` be `None` when the container is represented as `optional`? 

Not actionable here since the IR is typed this way, but I wonder if we should have made this required?

```yaml
  ExampleOptionalContainer:
    properties:
      optional: optional<ExampleTypeReference>
      valueType: TypeReference
```

Reply:
I guess it makes sense cause you want to make it easy to say you have an optional type, but no example? Agree that it is a bit odd though